src/sf_serialmac.h

```
@@ -188,7 +196,8 @@ enum sf_serialmac_return sf_serialmac_init ( struct sf_serialmac_ctx *ctx,
         SF_SERIALMAC_HAL_WRITE_FUNCTION write, SF_SERIALMAC_EVENT rx_event,
         SF_SERIALMAC_EVENT rx_buffer_event, SF_SERIALMAC_EVENT rx_sync_event,
         SF_SERIALMAC_EVENT tx_event, SF_SERIALMAC_EVENT tx_buffer_event,
-        SF_SERIALMAC_EVENT_ERROR error_event );
+        SF_SERIALMAC_EVENT_ERROR error_event,
+        bool useInvertedLengthField );
```

The init function should return an ERROR status if the library was not build with selectable inverted length field but the caller attempts to set it to the not default value.
Please add a new ERROR status to the **sf_serialmac_return** enums: **SF_SERIALMAC_RETURN_ERROR_UNSUPPORTED_PARAMETER**.
Otherwise the library will silently ignore the unsupported parameter without the user noticing it.

---

Add correct context header length initialization (See comment on **sf_serialmac_ctx**).

---

- [x] Done.